astropy/modeling/core.py

```diff
@@ -2880,6 +2880,10 @@ def _format_str(self, keywords=[], defaults={}):
             # Set units on the columns
             for name in self.param_names:
                 param_table[name].unit = getattr(self, name).unit
+                if self.name is not None:
+                    param_table[name].info.name = str(self.name + "_" + name)
+                else:
+                    param_table[name].info.name = name
```

Comment:
```suggestion
                param_table[name].info.name = name if self.name is None else f"{self.name}_{name}"
```

Coverage indicates these lines are not covered by tests, pleas check that your test actually covers them.